app/src/androidTest/java/com/aws/amazonlocation/ui/main/ExploreFragmentMapLanguageTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.aws.amazonlocation.ui.main
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import androidx.test.uiautomator.Until
+import com.aws.amazonlocation.*
+import com.aws.amazonlocation.di.AppModule
+import com.aws.amazonlocation.utils.*
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import org.hamcrest.CoreMatchers.allOf
+import org.junit.Assert
+import org.junit.Test
+
+@UninstallModules(AppModule::class)
+@HiltAndroidTest
+class ExploreFragmentMapLanguageTest : BaseTestMainActivity() {
+
+    private val uiDevice = UiDevice.getInstance(getInstrumentation())
+
+    private lateinit var preferenceManager: PreferenceManager
+
+    @Throws(java.lang.Exception::class)
+    override fun before() {
+        preferenceManager = PreferenceManager(ApplicationProvider.getApplicationContext())
+        preferenceManager.setValue(IS_APP_FIRST_TIME_OPENED, true)
+        preferenceManager.removeValue(KEY_MAP_NAME)
+        preferenceManager.removeValue(KEY_MAP_STYLE_NAME)
+        super.before()
+    }
+
+    @Test
+    fun testMapLanguageChangeTest() {
+        try {
+            Thread.sleep(DELAY_2000)
+            val btnContinueToApp = uiDevice.findObject(UiSelector().resourceId("${BuildConfig.APPLICATION_ID}:id/btn_continue_to_app"))
+            if (btnContinueToApp.exists()) {
+                btnContinueToApp.click()
+                Thread.sleep(DELAY_2000)
+            }
+            uiDevice.findObject(By.text(WHILE_USING_THE_APP))?.click()
+            uiDevice.findObject(By.text(WHILE_USING_THE_APP_CAPS))?.click()
+            uiDevice.findObject(By.text(WHILE_USING_THE_APP_ALLOW))?.click()
+            uiDevice.findObject(By.text(ALLOW))?.click()
+            Thread.sleep(DELAY_2000)
+            enableGPS(ApplicationProvider.getApplicationContext())
+            Thread.sleep(DELAY_2000)
+            uiDevice.wait(Until.hasObject(By.desc(AMAZON_MAP_READY)), DELAY_15000)
+
+            Thread.sleep(DELAY_2000)
+
+            goToMapStyles()
+
+            val clMapLanguage =
+                onView(withId(R.id.cl_map_language)).check(matches(isDisplayed()))
+            clMapLanguage.perform(click())
+
+            Thread.sleep(DELAY_2000)
+
+            val language =
+                waitForView(allOf(withText(TEST_WORD_LANGUAGE_AR), isDisplayed()))
+            language?.perform(click())
+
+            Thread.sleep(DELAY_2000)
```

Comment:
Are these 2 second sleeps the only way we can verify it is safe to move to the next step, or is there event-based mechanism?

Reply:
Changed it to `waitForView`